ui/webui/test/run-test

```diff
@@ -4,13 +4,36 @@ set -e
 
 clean_up () {
     ARG=$?
-    pkill -9 -f  'python3 -m http.server'
-    virsh -c qemu:///session destroy test-updates
-    virsh -c qemu:///session undefine test-updates
+    pkill -9 -f  "python3 -m http.server ${http_port}"
+    virsh -c qemu:///session destroy "fedora-35-127.0.0.2-${ssh_port}"
+    virsh -c qemu:///session undefine "fedora-35-127.0.0.2-${ssh_port}"
     exit $ARG
 }
 
 trap clean_up INT TERM EXIT
+
+# determine cockpit forward port
+http_port=8000
+until echo '' | nc 127.0.0.1 $http_port 2>&1 >/dev/null | grep -q 'Connection refused'; do
+    http_port=$((http_port+1))
+done
+
+# determine ssh forward port
+ssh_port=22000
+until echo '' | nc 127.0.0.2 $ssh_port 2>&1 >/dev/null | grep -q 'Connection refused'; do
+    ssh_port=$((ssh_port+1))
+done
+
+# determine cockpit forward port
+cockpit_port=9091
+until echo '' | nc 127.0.0.2 $cockpit_port 2>&1 >/dev/null | grep -q 'Connection refused'; do
+    cockpit_port=$((cockpit_port+1))
+done
```

Comment:
I wonder if there is no way to tell - just give me a free port?

Reply:
That would be really cool - like - give me a free port in that range - but I am not familiar with it.